# Construction of pYPK0_FBA1_ScTAL1_PDC1 (pMEC1130)


[pYPKa_Z_FBA1](pYPKa_Z_FBA1.ipynb)

[pYPKa_E_PDC1](pYPKa_E_PDC1.ipynb)

In [1]:
from pydna.all import *
from pydna import ipynb_importer

In [2]:
from ypk_std_primers import p166, p167, p567, p568

[Yeast Pathway Kit Standard Primers](ypk_std_primers.ipynb)

In [3]:
from Bio.Restriction import ZraI, AjiI, EcoRV

In [4]:
pYPK0 =read("pYPK0.gb")

In [5]:
promoter_clone = pYPKa_Z_FBA1 =read("pYPKa_Z_FBA1.gb")

[Genbank](http://www.ncbi.nlm.nih.gov/nucleotide/NM_001182243.1) record

In [6]:
from IPython.display import HTML
HTML('<iframe src=http://www.ncbi.nlm.nih.gov/pubmed/9169869 width=100% height=250></iframe>')

In [7]:
gb =Genbank("bjornjobbb@gmail.com")
gene_template = gb.nucleotide("NM_001182243.1")

In [8]:
gene_template.list_features()

+----------+-----------+-------+-------+--------+--------------+--------+------+
| Feature# | Direction | Start |  End  | Length | id           | type   | orf? |
+----------+-----------+-------+-------+--------+--------------+--------+------+
| 0        |    -->    |   0   |  1008 |   1008 | <unknown id> | source | yes  |
| 1        |    -->    |   <0  | >1008 |   1008 | <unknown id> | gene   | yes  |
| 2        |    -->    |   0   |  1008 |   1008 | <unknown id> | CDS    | yes  |
+----------+-----------+-------+-------+--------+--------------+--------+------+

In [9]:
orf = gene_template.extract_feature(2)

In [10]:
orf.isorf()

True

In [11]:
print(len(orf))
orf.seguid()

1008


GgJ9s-lK11rrZmXZDHjE3AjiVrs

In [12]:
print(str(orf.seq.translate()))

MSEPAQKKQKVANNSLEQLKASGTVVVADTGDFGSIAKFQPQDSTTNPSLILAAAKQPTYAKLIDVAVEYGKKHGKTTEEQVENAVDRLLVEFGKEILKIVPGRVSTEVDARLSFDTQATIEKARHIIKLFEQEGVSKERVLIKIASTWEGIQAAKELEEKDGIHCNLTLLFSFVQAVACAEAQVTLISPFVGRILDWYKSSTGKDYKGEADPGVISVKKIYNYYKKYGYKTIVMGASFRSTDEIKNLAGVDYLTISPALLDKLMNSTEPFPRVLDPVSAKKEAGDKISYISDESKFRFDLNEDAMATEKLSEGIRKFSADIVTLFDLIEKKVTA*


In [13]:
pf, pr =parse('''>TAL1_YPK_rec_fwd 
                        gtcgaggaacgccaggttgcccactttctcactagtgaAAATGTCTGAACCAGCTCAAAAGAA
                        >TAL1_YPK_rec_rev 
                        ATTTAAatcctgatgcgtttgtctgcacagatggcgcgTTAAGCGGTAACTTTCTTTTCAATCA''')

In [14]:
g =pcr( pf, pr, orf)

In [15]:
g.figure()

                                        5ATGTCTGAACCAGCTCAAAAGAA...TGATTGAAAAGAAAGTTACCGCTTAA3
                                                                   |||||||||||||||||||||||||| tm 56.8 (dbd) 65.6
                                                                  3ACTAACTTTTCTTTCAATGGCGAATTgcgcggtagacacgtctgtttgcgtagtcctaAATTTA5
5gtcgaggaacgccaggttgcccactttctcactagtgaAAATGTCTGAACCAGCTCAAAAGAA3
                                         ||||||||||||||||||||||| tm 58.0 (dbd) 65.3
                                        3TACAGACTTGGTCGAGTTTTCTT...ACTAACTTTTCTTTCAATGGCGAATT5

In [16]:
terminator_clone = pYPKa_E_PDC1 =read("pYPKa_E_PDC1.gb")

In [17]:
p  =pcr( p167, p567, promoter_clone)
t  =pcr( p568, p166, terminator_clone)

In [18]:
pYPK0_E_Z, stuffer = pYPK0.cut((EcoRV, ZraI))

In [19]:
(pYPK0_E_Z, p, g, t)

(Dseqrecord(-5681), Amplicon(792), Amplicon(1086), Amplicon(1061))

In [20]:
asm =Assembly((pYPK0_E_Z, p, g, t), limit=31)

In [21]:
asm

Assembly:
Sequences........................: [5681] [792] [1086] [1061]
Sequences with shared homologies.: [5681] [792] [1061] [1086]
Homology limit (bp)..............: 31
Number of overlaps...............: 4
Nodes in graph(incl. 5' & 3')....: 6
Only terminal overlaps...........: No
Circular products................: [8374]
Linear products..................: [8472] [8435] [8424] [8411] [7730] [7411] [7375] [6681] [6375] [2852] [2110] [1828] [98] [61] [50] [37]

In [22]:
candidate = asm.circular_products[0]
candidate.figure()

 -|pYKP0|98
|        \/
|        /\
|        98|792bp_PCR_prod|38
|                          \/
|                          /\
|                          38|1086bp_PCR_prod|35
|                                             \/
|                                             /\
|                                             35|1061bp_PCR_prod|61
|                                                                \/
|                                                                /\
|                                                                61-
|                                                                   |
 -------------------------------------------------------------------

In [23]:
pYPK0_FBA1_ScTAL1_PDC1 = candidate.synced(pYPK0)

The pYPK0_FBA1_ScTAL1_PDC1 has cseguid ```z56XvyucVzpo2iNPYy5TbXbxWlk``` and 8374 bp.

In [24]:
print(len(pYPK0_FBA1_ScTAL1_PDC1))
pYPK0_FBA1_ScTAL1_PDC1.cseguid()

8374


z56XvyucVzpo2iNPYy5TbXbxWlk

In [25]:
pYPK0_FBA1_ScTAL1_PDC1.name="pMEC1130"
#pYPK0_FBA1_ScTAL1_PDC1.description = "pYPK0_FBA1_ScTAL1_PDC1tp"
pYPK0_FBA1_ScTAL1_PDC1.stamp()

cSEGUID_z56XvyucVzpo2iNPYy5TbXbxWlk

In [26]:
pYPK0_FBA1_ScTAL1_PDC1.write("pYPK0_FBA1_ScTAL1_PDC1.gb")

pYPK0_FBA1_ScTAL1_PDC1.gb

##Download sequence

[pYPK0_FBA1_ScTAL1_PDC1](pYPK0_FBA1_ScTAL1_PDC1.gb)

In [27]:
from pydna.all import *

In [28]:
reloaded =read("pYPK0_FBA1_ScTAL1_PDC1.gb")

In [29]:
reloaded.cseguid()

z56XvyucVzpo2iNPYy5TbXbxWlk

In [30]:
reloaded.description

'cSEGUID_z56XvyucVzpo2iNPYy5TbXbxWlk_2015-07-15T12:32:52.717518'